backend/src/middleware/auth.ts

```diff
@@ -0,0 +1,40 @@
+import jwt, { JwtPayload } from "jsonwebtoken";
+import * as dotenv from "dotenv";
+import { Request as ExpressRequest, Response, NextFunction } from "express";
+
+dotenv.config();
+
+const { JWT_SECRET } = process.env;
+
+interface AuthRequest extends ExpressRequest {
+  userId?: string;
+}
+interface TokenPayload extends JwtPayload {
+  userId?: string;
+}
+
+export const isAuth = (req: AuthRequest, res: Response, next: NextFunction) => {
+  const authHeader = req.get("Authorization");
+  if (!authHeader) {
+    return res.status(401).json({
+      message: "Not authenticated",
+    });
+  }
+  const token = authHeader.split(" ")[1];
```

Comment:
```js
const token = authHeader.split(" ")[1];
```
This can return `undefined` if `authHeader.split(" ")` returns an array with length that 2.

Reply:
Please add a check to verify if token value is `undefined/null` & throw appropriate error.